server/src/main/java/org/elasticsearch/health/node/DiskHealthIndicatorService.java

```diff
@@ -0,0 +1,284 @@
+                    .stream()
+                    .filter(entry -> HealthStatus.RED.equals(entry.getValue().healthStatus()))
+                    .map(Map.Entry::getKey)
+                    .collect(Collectors.toSet());
+                problemIndices = Stream.concat(getIndicesForNodes(problemNodes).stream(), indicesWithBlock.stream())
+                    .collect(Collectors.toSet());
+                symptom = String.format(
+                    Locale.ROOT,
+                    "%d node%s out of disk space.%s",
+                    problemNodes.size(),
+                    problemNodes.size() > 1 ? "s are" : " is",
+                    problemIndices.size() > 0
+                        ? String.format(
+                            Locale.ROOT,
+                            " As a result %d ind%s cannot process any more updates.",
+                            problemIndices.size(),
+                            problemIndices.size() > 1 ? "ices" : "ex"
+                        )
+                        : ""
+                );
+                if (problemIndices.size() > 1) {
+                    impacts.add(RED_HEALTH_AFFECTED_INDICES_INGEST_IMPACT);
+                }
+            } else {
+                problemNodes = diskHealthInfoMap.entrySet()
+                    .stream()
+                    .filter(entry -> HealthStatus.YELLOW.equals(entry.getValue().healthStatus()))
+                    .map(Map.Entry::getKey)
+                    .collect(Collectors.toSet());
+                problemIndices = getIndicesForNodes(problemNodes);
+                symptom = String.format(
+                    Locale.ROOT,
+                    "%d node%s increased disk space.%s",
+                    problemNodes.size(),
+                    problemNodes.size() > 1 ? "s have" : " has",
+                    problemIndices.size() > 0
+                        ? String.format(
+                            Locale.ROOT,
+                            " As a result %d ind%s at risk of not being able to process any more updates.",
+                            problemIndices.size(),
+                            problemIndices.size() > 1 ? "ices are" : "ex is"
+                        )
+                        : ""
+                );
+                if (problemIndices.size() > 1) {
+                    impacts.add(YELLOW_HEALTH_AFFECTED_INDICES_INGEST_IMPACT);
+                }
+            }
+            Set<DiscoveryNodeRole> roles = clusterService.state()
+                .nodes()
+                .getNodes()
+                .values()
+                .stream()
+                .filter(node -> problemNodes.contains(node.getId()))
+                .map(DiscoveryNode::getRoles)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toSet());
+            if (roles.contains(DiscoveryNodeRole.MASTER_ROLE)) {
+                impacts.addAll(StableMasterHealthIndicatorService.UNSTABLE_MASTER_IMPACTS);
```

Comment:
Do we want to report all these? I think one impact that summarises the impact on the master node would be enough? If we decide to have the same impacts (all from the stable master indicator) I think we shouldn't create this coupling between indicators but redefine the impacts in this indicator.

What do you think? (@gmarouli)

Reply:
I agree, so I would expect that if the disk is actually causing the master to be unstable, then the unstable master will elaborately say the impact. Here we "assume" that a master running out of disk will cause issues, but we do not really know. I would suggest just saying that the master is running out of disk and that could have negative impact on the following areas is enough. What do you think?

Reply:
What are we going to do if the master and data nodes are fine but the ingest node or ml node are full? We wouldn't report the master node impacts in that case right? But we also wouldn't want to just ignore those would we?

Reply:
>ingest node or ml node are full? We wouldn't report the master node impacts in that case right?

That's correct.

>But we also wouldn't want to just ignore those would we?

I assume "those" here are the possible impacts of ingest/ML nodes being full? If so, yes, we should NOT ignore them but report ingest/ML related impacts.
If "those" meant the "master related impacts", I think we should ignore them as the master nodes are fine.